[003.2] Finished code for the computer's turn | H.C.

iobosse03 opened the pull request:
003.2 Schere Stein Papier https://app.asana.com/0/1205072348884926/1205082265132033/f

Contains code for 003.3 that is commented out

iobosse03 committed:
[003] rock paper scissors finished; code cleaning | I.B., J.H.

iobosse03 committed:
[003] removed a line used for debugging; code cleaning | I.B.

iobosse03 committed:
[003.2] added comments to indicate which functions are not yet used

iobosse03 committed:
Merge branch '003.1' into 003.2Javascript

iobosse03 committed:
[003.2] moved eventlisteners from rock-paper-scissors.js to game.js, gave them the proper functions

iobosse03 committed:
[003.2] Updated the numbers from 0-2 to correspond to the english variation of rock paper scissors

iobosse03 committed:
[003.2] Commented out code not needed for this story

Ready for review

SpaTeamAcademy reviewed (changes requested):
Clean Coding || Leserlichkeit des Codes

-> Klicken des Buttons könnte auch gleich "Rock", "Paper" oder "Scissor" als Parameter übergeben, damit muss nicht zwischen Zahlen und Text konvertiert werden

-> Warum wird zwischen "Rock" und "rock" unterschieden? Das verdoppelt lediglich den Code in der function toText() || DRY-Prinzip

-> sollte Groß-/Kleinschreibung zwingend notwendig sein, seht euch die entspechenden string-Methoden dazu an (.toUpperCase(), .toLowerCase())

iobosse03 commented:
Sollen alle Funktionen, die die Zahlen verwenden auf Strings geändert werden, oder nur die toText() Funktion?

SpaTeamAcademy commented:
Für die Leserlichkeit würde ich empfehlen Strings statt Zahlen zu verwenden. Ihr spart euch dadurch einerseits erklärende Kommentare, welche Zahlen wofür stehen. Weiters könnt ihr mit strings leichter und verständlicher arbeiten, wenn der Code z.B. nachträglich geändert werden muss (Ihr müsst euch dann auch nicht selbst immer wieder überlegen, wofür zB. die 1 steht)

Im weiteren Verlauf führt das zu weniger Verwirrung bei den Logikoperationen ("rock" == "rock") und Gewinnbedingungen (0 = Rock und 0 = Draw). 

Es sollten nur dann Zahlen verwendet werden, wenn dies für die Programmieraufgabe notwendig ist (z.B. Zählen von Werten, Rechenaufgaben,...)

iobosse03 committed:
[003.2] changed eventlisteners to call functions using strings as parameters

003.2 Schere Stein Papier (Computer spielt) https://app.asana.com/0/1205072348884926/1205082265132033/f

iobosse03 committed:
[003.2] changed functions to correctly use the string parameters

003.2 Schere Stein Papier (Computer spielt) https://app.asana.com/0/1205072348884926/1205082265132033/f

iobosse03 committed:
[003.2] removed the now unneccessary toText() function

003.2 Schere Stein Papier (Computer spielt) https://app.asana.com/0/1205072348884926/1205082265132033/f

iobosse03 commented:
Code review: J.H.
